tasks/request_status_for_job/request_status_for_job.py

```diff
@@ -0,0 +1,164 @@
+        OUTPUT_GRANULES_KEY: status_entries
+    }
+
+
+def get_granule_status_entries_for_job(job_id: str, db_connect_info: Dict) -> List[Dict[str, Any]]:
+    """
+    Gets the orca_recoveryjob status entry for the associated job_id.
+
+    Args:
+        job_id: The unique asyncOperationId of the recovery job to retrieve status for.
+        db_connect_info: The {database} defined db_connect_info.
+
+    Returns: A list of dicts with the following keys:
+        'granule_id' (str)
+        'status' (str): pending|staged|success|failed
+
+    """
+    sql = f"""
+            SELECT
+                granule_id as \"{OUTPUT_GRANULE_ID_KEY}\",
+                orca_status.value AS \"{OUTPUT_STATUS_KEY}\"
+            FROM
+                orca_recoveryjob
+            JOIN orca_status ON orca_recoveryjob.status_id=orca_status.id
+            WHERE
+                job_id = %s
+            """
+
+    try:
+        rows = database.single_query(sql, db_connect_info, (job_id,))
+    except database.DbError as err:
+        LOGGER.error(f"DbError: {str(err)}")
+        raise DatabaseError(str(err))
+    result = database.result_to_json(rows)
+    return result
+
+
+def get_status_totals_for_job(job_id: str, db_connect_info: Dict) -> Dict[str, int]:
+    # noinspection SpellCheckingInspection
+    """
+    Gets the number of orca_recoveryjobs for the given job_id for each possible status value.
+
+    Args:
+        job_id: The unique id of the recovery job to retrieve status for.
+        db_connect_info: The database defined db_connect_info.
+
+    Returns: A dictionary with the following keys:
+        'pending' (int)
+        'staged' (int)
+        'success' (int)
+        'failed' (int)
+    """
+    # noinspection SpellCheckingInspection
+    sql = f"""
+            with granule_status_count AS (
+                SELECT status_id
+                    , count(*) as total
+                FROM orca_recoveryjob
+                WHERE job_id = %s
+                GROUP BY status_id
+            )
+            SELECT value
+                , coalesce(total, 0) as total
+            FROM orca_status os
+            LEFT JOIN granule_status_count gsc ON (gsc.status_id = os.id)"""
+
+    try:
+        rows = database.single_query(sql, db_connect_info, (job_id,))
+    except database.DbError as err:
+        LOGGER.error(f"DbError: {str(err)}")
+        raise DatabaseError(str(err))
+    # todo: retest
+    totals = {rows[i]['value']: rows[i]['total'] for i in range(0, len(rows), 1)}
+    return totals
+
+
+def create_http_error_dict(error_type: str, http_status_code: int, request_id: str, message: str) -> Dict[str, Any]:
+    LOGGER.error(message)
+    return {
+        'errorType': error_type,
+        'httpStatus': http_status_code,
+        'requestId': request_id,
+        'message': message
+    }
+
+
+def handler(event: Dict[str, Any], context: Any) -> Dict[str, Any]:
+    """
+    Entry point for the request_status_for_job Lambda.
```

Comment:
Add additional documentation on environment variables needed. At a minimum you would need prefix, host, port, username, password.